testnet/stacks-node/src/globals.rs

```diff
@@ -0,0 +1,289 @@
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::mpsc::SyncSender;
+use std::sync::{Arc, Mutex};
+
+use stacks::burnchains::Txid;
+use stacks::chainstate::burn::operations::LeaderKeyRegisterOp;
+use stacks::chainstate::burn::BlockSnapshot;
+use stacks::chainstate::coordinator::comm::CoordinatorChannels;
+use stacks::chainstate::stacks::db::unconfirmed::UnconfirmedTxMap;
+use stacks::chainstate::stacks::db::StacksChainState;
+use stacks::chainstate::stacks::miner::MinerStatus;
+use stacks::net::NetworkResult;
+use stacks_common::types::chainstate::{BlockHeaderHash, BurnchainHeaderHash, ConsensusHash};
+
+use crate::neon::Counters;
+use crate::neon_node::LeaderKeyRegistrationState;
+use crate::run_loop::RegisteredKey;
+use crate::syncctl::PoxSyncWatchdogComms;
+
+pub type NeonGlobals = Globals<RelayerDirective>;
+
+/// Command types for the relayer thread, issued to it by other threads
+pub enum RelayerDirective {
+    /// Handle some new data that arrived on the network (such as blocks, transactions, and
+    HandleNetResult(NetworkResult),
+    /// Announce a new sortition.  Process and broadcast the block if we won.
+    ProcessTenure(ConsensusHash, BurnchainHeaderHash, BlockHeaderHash),
+    /// Try to mine a block
+    RunTenure(RegisteredKey, BlockSnapshot, u128), // (vrf key, chain tip, time of issuance in ms)
+    /// A nakamoto tenure's first block has been processed.
+    NakamotoTenureStartProcessed(ConsensusHash, BlockHeaderHash),
+    /// Try to register a VRF public key
+    RegisterKey(BlockSnapshot),
+    /// Stop the relayer thread
+    Exit,
+}
+
+/// Inter-thread communication structure, shared between threads. This
+/// is generic over the relayer communication channel: nakamoto and
+/// neon nodes use different relayer directives.
+pub struct Globals<T> {
+    /// Last sortition processed
+    last_sortition: Arc<Mutex<Option<BlockSnapshot>>>,
+    /// Status of the miner
+    miner_status: Arc<Mutex<MinerStatus>>,
+    /// Communication link to the coordinator thread
+    pub(crate) coord_comms: CoordinatorChannels,
+    /// Unconfirmed transactions (shared between the relayer and p2p threads)
+    unconfirmed_txs: Arc<Mutex<UnconfirmedTxMap>>,
+    /// Writer endpoint to the relayer thread
+    pub relay_send: SyncSender<T>,
+    /// Cointer state in the main thread
+    pub counters: Counters,
+    /// Connection to the PoX sync watchdog
+    pub sync_comms: PoxSyncWatchdogComms,
```

Comment:
This can be dropped for Nakamoto as well -- the system _must_ wait for the missing PoX anchor block in all cases.

Reply:
Same as above-- this struct is shared with the `neon_node`. This field eventually will need to be generic. I think if we don't need the `pox_watchdog`, this would just be a struct monitoring the `mine_after` setting (where miners can set when they want to begin issuing block commits).